### PR TITLE
Database to xml

### PR DESCRIPTION
**Pull request by AlexDeweert**
logging basically works, might need some tweaks when we put it on lab machines

### CODE REVIEW

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class LogController {
+    private LoggingService loggingService;
+
+    public LogController(LoggingService loggingService) {
+        this.loggingService = loggingService;
+    }
+
+    @RequestMapping(value = "/printlogs")
+    public void printAllLogs() {
+        System.out.println("Attempting to print logs based on call to \"printAllLogs()\"");
```
Comment (BenDHillier) [inline StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java]: remove print

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+@RestController
+public class LogController {
+    private LoggingService loggingService;
+
+    public LogController(LoggingService loggingService) {
+        this.loggingService = loggingService;
+    }
+
+    @RequestMapping(value = "/printlogs")
```
Comment (BenDHillier) [inline StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java]: Is this command just for debugging?

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/SellController.java [hunk 3]
```diff
@@ -45,61 +46,55 @@ Quote createNewSell(
...
-                        .build());
-
-        if (amount <= 0) {
-            throw new IllegalArgumentException(
-                    "The amount parameter must be greater than zero.");
-        }
-
         Quote quote = quoteService.getQuote(stockSymbol, userId);
+        try {
+            //loggingService.logUserCommand(CommandType.SELL,userId,stockSymbol,null,amount);
```
Comment (BenDHillier) [inline StockTrader/src/main/java/com/restResource/StockTrader/controller/SellController.java]: was there a problem having the amount here?

File: StockTrader/src/main/resources/schema.sql [hunk 4]
```diff
@@ -37,45 +37,48 @@ CREATE TABLE account (
 );
 
 CREATE TABLE account_transaction_log (
-    action varchar,
+    transaction_num SERIAL PRIMARY KEY,
+    action varchar(255),
     funds integer,
-    timestamp timestamp,
-    username varchar
+    timestamp bigint,
```
Comment (BenDHillier) [inline StockTrader/src/main/resources/schema.sql]: why is this changed to bigint?

File: StockTrader/src/main/resources/schema.sql [hunk 5]
```diff
@@ -37,45 +37,48 @@ CREATE TABLE account (
...
-      ELSIF (OLD.amount > NEW.amount) THEN
+        funds = New.amount - old_amount;
+      ELSIF (old_amount > NEW.amount) THEN
         action = ''remove'';
-        funds = OLD.amount - NEW.amount;
+        funds = old_amount - NEW.amount;
       ELSE
         RETURN NULL;
       END IF;
-      INSERT INTO account_transaction_log (action, funds, timestamp, username)
```
Comment (BenDHillier) [inline StockTrader/src/main/resources/schema.sql]: We should still insert into account transaction log, since we need transaction history for the display_summary command

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/AddController.java [hunk 6]
```diff
@@ -24,24 +38,19 @@ public AddController(AccountRepository accountRepository, LoggingService logging
...
-            }
-            else {
+            } else {
                 accountRepository.updateAccountBalance(userId, amount);
             }
             return HttpStatus.OK;
-        } catch( IllegalArgumentException e ) {
+        } catch (Exception e) {
             System.out.println("Exception in AddController: " + e.toString());
+            loggingService.logErrorEvent(CommandType.ADD,userId,null,null,amount,"Amount added must be lteq 0");
```
Comment (BenDHillier) [inline StockTrader/src/main/java/com/restResource/StockTrader/controller/AddController.java]: This would be a great place to use a builder type. Would avoid the multiple null params, and would be much more readable and less error prone (easy to get the order mixed up here). It's fine for now, we have a deadline, but I'd like it changed in the future.

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java (hunk 1 above)
Comment (AlexDeweert) [reply StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java]: done

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java (hunk 2 above)
Comment (AlexDeweert) [reply StockTrader/src/main/java/com/restResource/StockTrader/controller/LogController.java]: Yeah it was. Removed.

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/SellController.java (hunk 3 above)
Comment (AlexDeweert) [reply StockTrader/src/main/java/com/restResource/StockTrader/controller/SellController.java]: no. amount added.

File: StockTrader/src/main/resources/schema.sql (hunk 4 above)
Comment (AlexDeweert) [reply StockTrader/src/main/resources/schema.sql]: Before it was timestamp, but we need to log the timestamp in xml output as unix time milliseconds. The xml schema that she checks against checks to see if the timestamp values are between a range of unix milliseconds. This was just a way to get the milliseconds into the log instead of a timestamp mont-day-year type of value.

File: StockTrader/src/main/resources/schema.sql (hunk 5 above)
Comment (AlexDeweert) [reply StockTrader/src/main/resources/schema.sql]: Wouldn't the XML dumplog_per_user file be enough? We could display the summary to the user (I assume  in a readable format on the client interface) based on the XML file data itself.

Comment (BenDHillier) [reply StockTrader/src/main/resources/schema.sql]: we don't want all the other log types mixed in though, the dumplog command downloads a file, and the dumplog command is probably going to be quite slow compared to what we would want for display_summary. Think of display_summary like going to your profile page. do you want a website to download an xml file every time you go to your profile?

File: StockTrader/src/main/java/com/restResource/StockTrader/controller/AddController.java (hunk 6 above)
Comment (AlexDeweert) [reply StockTrader/src/main/java/com/restResource/StockTrader/controller/AddController.java]: Sure. I tried to continue to use builder but I was getting null object types. But yeah you're right, builder form is nicer.

File: StockTrader/src/main/resources/schema.sql (hunk 5 above)
Comment (AlexDeweert) [reply StockTrader/src/main/resources/schema.sql]: Okay. What do you think we would want for display summary? It doesn't say very much on the project website.
